File: RestonetBackend_version2/RestonetBackend_version2/src/main/java/ma/inetum/restonetbackend/entities/Article.java
package ma.inetum.restonetbackend.entities;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "ARTICLES")
@NoArgsConstructor
@AllArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Article implements Serializable {
    @Id
    @Column(name = "ARTICLE")
    private String ARTICLE;
    @Column(name = "LIBELLECOURT")
    private String LIBELLECOURT;
    @Column(name = "NOM")
    private String NOM;
    @Column(name = "FAMILLE")
    private String FAMILLE;
    @Column(name = "PRIXVENTEATVA1")
    private Double PRIXVENTEATVA1;
    @Column(name = "PRIXVENTEATVA2")
    private Double PRIXVENTEATVA2;
    @Column(name = "PRIXVENTEATVA3")
    private Double PRIXVENTEATVA3;
    @OneToMany(mappedBy = "article", fetch = FetchType.EAGER)
    private List<DetailRepasAReservation> repasDetails;


}

File: reco-administrateurFront_version2/reco-administrateurFront_version2/src/app/models/article.ts
import { DetailRepasAReservation } from "./detail-repas-areservation";

export class Article {
  article?: string;
  libelleCourt?: string;
  nom?: string;
  famille?: string;
  prixventeAtva1?: string;
  prixVenteAtva2?: string;
  prixVenteAtva3?: string;
  repasDetail:Array<DetailRepasAReservation>= [];
}
